engine/runtime/src/main/scala/org/enso/compiler/pass/analyse/AliasAnalysis.scala

```diff
@@ -554,21 +612,24 @@ case object AliasAnalysis extends IRPass {
     */
   def analyseName(
     name: IR.Name,
+    bindings: BindingsMap,
     isInPatternContext: Boolean,
     graph: Graph,
     parentScope: Scope
   ): IR.Name = {
     val occurrenceId = graph.nextId()
 
-    if (isInPatternContext && name.isVariable) {
+    val resolution = bindings.resolveName(name.name)
+    val isVariable = !name.isMethod && !isTpeConstructor(resolution)
```

Comment:
I'm very uneasy reading this. What does it matter for alias analysis that something resolved to `ResolvedConstructor`? It shouldn't. Thinking about it even deeper – why does `AliasAnalysis` care about global resolution at all? It should just always try to bind the name to a local symbol and if it fails assume it's global. The old version of this was using a syntactic distinction, we need to carefully think about the new one. Unless I'm missing something, but from my knowledge, it makes no logical sense for the condition to be what it is.

Reply:
Yea, this was unnecessary as it turned out. I was mimicking the old logic and it turned out to be an easier way.